addons/io/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/NeeoUtil.java

```diff
@@ -0,0 +1,290 @@
+     * @return the servlet url
+     */
+    public static String getServletUrl(String brainId) {
+        requireNotEmpty(brainId, "brainId cannot be empty");
+        return NeeoConstants.WEBAPP_PREFIX + "/" + brainId;
+    }
+
+    /**
+     * Gets the unique identifier of this openHAB instance
+     *
+     * @return the unique id
+     */
+    public static String getUniqueId() {
+        return InstanceUUID.get();
+    }
+
+    /**
+     * Decodes the passed UTF-8 String using an algorithm that's compatible with
+     * JavaScript's <code>decodeURIComponent</code> function. Returns
+     * <code>null</code> if the String is <code>null</code>.
+     *
+     * @param s The UTF-8 encoded String to be decoded
+     * @return the decoded String
+     */
+    public static String decodeURIComponent(String s) {
+        if (s == null) {
+            return null;
+        }
+
+        String result = null;
+
+        try {
+            result = URLDecoder.decode(s, StandardCharsets.UTF_8.name());
+        }
+
+        // This exception should never occur.
+        catch (UnsupportedEncodingException e) {
+            result = s;
+        }
+
+        return result;
+    }
+
+    /**
+     * Encodes the passed String as UTF-8 using an algorithm that's compatible
+     * with JavaScript's <code>encodeURIComponent</code> function. Returns
+     * <code>null</code> if the String is <code>null</code>.
+     *
+     * @param s The String to be encoded
+     * @return the encoded String
+     */
+    public static String encodeURIComponent(String s) {
+        requireNotEmpty(s, "s cannot be null or empty");
+        String result = null;
+
+        try {
+            result = URLEncoder.encode(s, "UTF-8").replaceAll("\\+", "%20").replaceAll("\\%21", "!")
+                    .replaceAll("\\%27", "'").replaceAll("\\%28", "(").replaceAll("\\%29", ")")
+                    .replaceAll("\\%7E", "~");
+        }
+
+        // This exception should never occur.
+        catch (UnsupportedEncodingException e) {
+            result = s;
+        }
+
+        return result;
+    }
+
+    /**
+     * Write a response out to the {@link HttpServletResponse}
+     *
+     * @param resp the non-null {@link HttpServletResponse}
+     * @param str the possibly null, possibly empty string content to write
+     * @throws IOException Signals that an I/O exception has occurred.
+     */
+    public static void write(HttpServletResponse resp, String str) throws IOException {
+        Objects.requireNonNull(resp, "resp cannot be null");
+
+        final Logger logger = LoggerFactory.getLogger(NeeoUtil.class);
+
+        resp.setContentType("application/json");
+        resp.setCharacterEncoding("UTF-8");
+        final PrintWriter pw = resp.getWriter();
+        if (StringUtils.isEmpty(str)) {
+            pw.print("{}");
+        } else {
+            pw.print(str);
+        }
+
+        logger.trace("Sending: {}", str);
+        pw.flush();
+        pw.close();
```

Comment:
Is this actually your responsibility, do all filters assume the writer is already closed?

Reply:
Sorry - old school and can't seem to get over needing to close things ;)